ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb

```diff
@@ -10,12 +10,21 @@
     Private dctKeyControls As Dictionary(Of String, ucrBaseDataLink)
 
     Public Overrides Sub PopulateControl()
-        MyBase.PopulateControl()
+        ' This is the cause of slow loading - getting all records into dtbRecords is slow.
+        'MyBase.PopulateControl()
+
+        ' Instead of getting the full table just get the number of rows.
+        ' This should work but doesn't for some reason.
+        'iMaxRows = CallByName(CallByName(clsDataConnection.db, clsDataDefinition.GetTableName(), CallType.Get), "Count", CallType.Method)
+        ' Tried this as an alternative too but gives same error
+        'Dim objTemp As Object = CallByName(clsDataConnection.db, clsDataDefinition.GetTableName(), CallType.Get)
+        'iMaxRows = CallByName(objTemp, "Count", CallType.Method)
+        ' Doing this to test with form_daily2 until work out how to do in general.
+        iMaxRows = clsDataConnection.db.form_daily2.Count()
```

Comment:
@volloholic @dannyparsons this will work. It turns out as @volloholic mentioned above that we needed to cast the object to an appropriate type that implements the count method. 
```
Dim x = CallByName(clsDataConnection.db, clsDataDefinition.GetTableName(), CallType.Get)
Dim y = TryCast(x, IQueryable(Of Object))
iMaxRows = y.Count()
```

We can easily move that to the data call.

Reply:
Thanks that worked.